src/frame/decompress.rs

```diff
@@ -399,3 +398,31 @@ impl<R: fmt::Debug + io::Read> fmt::Debug for FrameDecoder<R> {
             .finish()
     }
 }
+
+/// Similar to set_len but panics if the vec doesn't have enough capacity.
+#[cfg(feature = "safe-decode")]
+#[inline]
+fn vec_set_len(v: &mut Vec<u8>, new_len: usize) {
+    debug_assert!(new_len <= v.capacity());
+    v.resize(new_len, 0);
+}
+
+/// Similar to set_len but panics if the vec doesn't have enough capacity.
+#[cfg(not(feature = "safe-decode"))]
+#[inline]
+fn vec_set_len(v: &mut Vec<u8>, new_len: usize) {
+    assert!(new_len <= v.capacity());
+    unsafe {
+        v.set_len(new_len);
+    }
+}
+
+/// Similar to `v.get_mut(start..end) but will adjust the len if needed.
+/// Panics if there's not enough capacity.
+#[inline]
+fn vec_bytes_mut(v: &mut Vec<u8>, start: usize, end: usize) -> &mut [u8] {
```

Comment:
`resize_and_get_mut` would be a better name I think

Reply:
Indeed, thanks.